cpp/src/arrow/memory_pool.h

```diff
@@ -180,6 +180,12 @@ Status jemalloc_set_decay_ms(int ms);
 /// May return NotImplemented if mimalloc is not available.
 ARROW_EXPORT Status mimalloc_memory_pool(MemoryPool** out);
 
+ARROW_EXPORT Status jemalloc_mallctl(const char* name, void* oldp, size_t* oldlenp,
```

Comment:
nit: docstrings, and group with the jemalloc function above?

Reply:
I'm not sure, but I would favour exposing higher-level APIs rather than this raw C ugliness.

Reply:
For example:
```c++
Status jemalloc_get_stat(const char* name, bool* out);
Status jemalloc_get_stat(const char* name, int64_t* out);
Status jemalloc_get_stat(const char* name, uint64_t* out);
Status jemalloc_get_stat(const char* name, const char** out);
```
(passing the result as out-parameter to allow for several signatures; we could also return a variant but that would add a heavy header inclusion)

Reply:
Higher level API does make sense. I suppose we also want to provide some tests with useful examples.

Reply:
Changed. I've gone with a subset as I don't think we'll need `int64_t` and `int64_t` for now.
Should these be `Status` or `Result`?